### PR TITLE
📖 docs: link Kubernetes contributor guide in CONTRIBUTING.md

### LINKED ISSUE #8061: Consider linking the Kubernetes New Contributor Course in our book
Kubernetes recently added a New Contributor Course to their book: https://www.kubernetes.dev/docs/onboarding/

While it is targeted at k/k development I think ~80% of it is relevant for new CAPI contributors as well.

I wonder if we should link it (and probably a few more other pages from kuberntes.dev as well) in our CAPI book

/kind documentation

### PR DESCRIPTION
**Pull request by Sajiyah-Salat**
**What this PR does / why we need it**:
Kubernetes recently added a New Contributor Course to their book: https://www.kubernetes.dev/docs/onboarding/

While it is targeted at k/k development I think ~80% of it is relevant for new CAPI contributors as well.
So I added it as requested. 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8061

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Could you remove the link from the contents table and maybe add something in line 37 like:

"Contributors who are not used to working in the Kuberntes ecosystem should also take a look at the Kubernetes [New Contributor Course](https://www.kubernetes.dev/docs/onboarding/).

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +34,7 @@
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
 Read the following guide if you're interested in contributing to cluster-api.
+Contributors who are not used to working in the Kuberntes ecosystem should also take a look at the Kubernetes [New Contributor Course.](https://www.kubernetes.dev/docs/onboarding/)
```
Comment (killianmuldoon) [inline CONTRIBUTING.md:37]:
```suggestion
Contributors who are not used to working in the Kubernetes ecosystem should also take a look at the Kubernetes [New Contributor Course.](https://www.kubernetes.dev/docs/onboarding/)
```

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
/retitle 📖docs: link Kubernetes contributor guide in CONTRIBUTING.md 

/lgtm

/assign @oscr

**Comment by oscr** [comment]
@Sajiyah-Salat Thank you for your contribution! Could you please squash the commits 3 commits into 1 as described [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#contributing-a-patch).

**Comment by Sajiyah-Salat** [comment]
Sure.

**Comment by Sajiyah-Salat** [comment]
I have squashed my commits.

**Comment by killianmuldoon** [comment]
Hi @Sajiyah-Salat  - there should only be one commit once you're finished squashing. Could you try to follow the squashing process again?

**Comment by Sajiyah-Salat** [comment]
I did some mistakes hence I am closing thi pr and starting a new on e with same changes. Please review #8288

**Pull request closed without merging**